File: chapters/sorting_searching/bubble/code/haskell/bubbleSort.hs
bubbleSort :: (Ord a) => [a] -> [a]
bubbleSort x = (!!length x) $ iterate bubble x
  where bubble (x:y:r)
          | x <= y    = x : bubble (y:r)
          | otherwise = y : bubble (x:r)
        bubble x = x

File: chapters/sorting_searching/bubble/code/cs/BubbleSort.cs
// submitted by Julian Schacher (jspp)
using System;
using System.Collections.Generic;

namespace BubbleSort
{
    public static class BubbleSort
    {
        public static List<T> RunBubbleSort<T>(List<T> list) where T : IComparable<T>
        {
            var length = list.Count;

            for (int i = 0; i < length; i++)
            {
                for (int j = 1; j < length; j++)
                {
                    if (list[j - 1].CompareTo(list[j]) > 0)
                    {
                        var temp = list[j - 1];
                        list[j - 1] = list[j];
                        list[j] = temp;
                    }
                }
            }

            return list;
        }
    }
}
